server/log_operation_manager.go

```diff
@@ -8,87 +8,106 @@ import (
 	"github.com/google/trillian/util"
 )
 
-// logOperationManager controls scheduling activities for logs. At the moment it's very simple
-// with a single task running over active logs one at a time. This will be expanded later.
-// This is meant for embedding into the actual operation implementations and should not
-// be created separately.
-type logOperationManager struct {
+// LogOperation defines a task that operates on logs. Examples are scheduling, signing,
+// consistency checking or cleanup.
+type LogOperation interface {
+	Name() string
+	ExecutePass([]trillian.LogID, LogOperationManagerContext) bool
+}
+
+// LogOperationManagerContext bundles up the values so testing can be made easier
+type LogOperationManagerContext struct {
 	// done is a channel that provides an exit signal
 	done chan struct{}
```

Comment:
I hoped so but it's really for request scoped things and it seemed to be a bit of a hack for long lived operations.